projects/components/src/combo-box/combo-box.component.ts

```diff
@@ -210,25 +252,63 @@ export class ComboBoxComponent<TValue = string> implements AfterViewInit, OnChan
 
   public onEnter(): void {
     if (this.arePopoverOptionsAvailable()) {
-      this.setText(this.filteredOptions[this.highlightedOptionIndex].text);
+      if (this.highlightedOptionIndex >= 0) {
+        this.setText(this.filteredOptions[this.highlightedOptionIndex].text);
+      } else {
+        this.setText(this.text);
+      }
       this.hidePopover();
     }
     this.enter.emit(this.buildResult());
   }
 
   public onNextOption(event: KeyboardEvent): void {
-    if (this.arePopoverOptionsAvailable()) {
+    event.preventDefault(); // Prevent tabbing to next control on page
+
+    if (!this.arePopoverOptionsAvailable()) {
+      return;
+    }
+
+    if (this.isCreateOptionAvailable() && this.isNextOptionBoundaryIndex()) {
+      if (this.highlightedOptionIndex < 0) {
+        this.highlightedOptionIndex = 0;
+      } else if (this.highlightedOptionIndex >= this.filteredOptions.length - 1) {
+        this.highlightedOptionIndex = -1;
+      }
+    } else {
       this.highlightedOptionIndex = ++this.highlightedOptionIndex % this.filteredOptions.length;
-      event.preventDefault(); // Prevent tabbing to next control on page
     }
+
+    this.scrollIntoViewService.scrollIntoView(this.optionElements.get(this.highlightedOptionIndex)?.nativeElement);
```

Comment:
Nice. Is it this easy to scroll and show a particular element in the view? I can use it at so many other places now.

Reply:
Yep! Just this easy.